### PR TITLE
8297088: Update LCMS to 2.14

### PR DESCRIPTION
**Pull request by alisenchung**
Redo of 8295369: Update LCMS to 2.14
Update LCMS to newest release.
Updated DISABLED_WARNINGS_gcc for LCMS to include type limits to prevent a build error in linux.
Automatic tests and J2D demo tests were run.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297088](https://bugs.openjdk.org/browse/JDK-8297088): Update LCMS to 2.14

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11217/head:pull/11217` \
`$ git checkout pull/11217`

Update a local copy of the PR: \
`$ git checkout pull/11217` \
`$ git pull https://git.openjdk.org/jdk pull/11217/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11217`

View PR using the GUI difftool: \
`$ git pr show -t 11217`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11217.diff">https://git.openjdk.org/jdk/pull/11217.diff</a>

</details>

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]
Build change looks good.

File: make/modules/java.desktop/lib/Awt2dLibraries.gmk [hunk 1]
```diff
@@ -305,7 +305,7 @@ $(eval $(call SetupJdkLibrary, BUILD_LIBLCMS, \
         common/awt/debug \
         libawt/java2d, \
     HEADERS_FROM_SRC := $(LIBLCMS_HEADERS_FROM_SRC), \
-    DISABLED_WARNINGS_gcc := format-nonliteral stringop-truncation, \
+    DISABLED_WARNINGS_gcc := format-nonliteral type-limits stringop-truncation, \
```
Comment (mrserb) [inline make/modules/java.desktop/lib/Awt2dLibraries.gmk:308]: it would be good to report this upstream.

**Review by mrserb — APPROVED** [review]

**Review by prrace — APPROVED** [review]

**Review by honkar-jdk — APPROVED** [review]

File: make/modules/java.desktop/lib/Awt2dLibraries.gmk (hunk 1 above)
Comment (mrserb) [reply make/modules/java.desktop/lib/Awt2dLibraries.gmk:308]: https://github.com/mm2/Little-CMS/issues/458

**Pull request closed without merging**